doc/flame/rendering/layers.md

```diff
@@ -1,4 +1,11 @@
-# Layers
+# Layers and Snapshots
+
+Layers and snapshots share some common features, including the ability to pre-render and cache
+objects for improved performance. However, they also have unique features which make them better
+suited for different use-cases.
```

Comment:
while I think both sections below do a good job of describing each feature in a vacuum, I think this paragraph could give more details about what are the differences and what rationale I should use to choose one or the other

Reply:
I've added a description. Please amend it if you don't agree!

To be honest, I'm not sure of when you might use a Layer instead of a Snapshot, unless you don't want the overhead of a position component. Layers support processors (of which there is only 1 so far, but more could be made). However processors are sort of doubling-up on `Effects`?

Reply:
# Layers and Snapshots

Layers and snapshots share some common features, including the ability to pre-render and cache
objects for improved performance. However, they also have unique features which make them better
suited for different use-cases.

`Snapshot` is a Mixin that can be added to any `PositionComponent`. Use this for:

- Mixing in to existing game objects (that are `PositionComponents`)
- Caching game objects, such as sprites, that are complex to render
- Drawing the same object many times without rendering it each time
- Capturing an image snapshot to save as a screenshot (for example)

`Layer` is a class. Use or extend this class for:

- Structuring your game with logical layers (e.g. UI, foreground, main, background)
- Grouping objects to form a complex scene, and then caching it (e.g. a background layer)
- Processor support. Layers allow user-defined processors to run pre- and post- render.

Reply:
Let's keep it like that for now and then we can extend on that in a follow-up. :)